tls/s2n_connection.h

```diff
@@ -95,6 +95,7 @@ struct s2n_connection {
 
     /* The PRF needs some storage elements to work with */
     union s2n_prf_working_space prf_space;
+    struct s2n_prf_implementation prf_impl;
```

Comment:
I think `struct s2n_prf_implementation *` makes more sense. Both here and all the subsequent ripples it will cause in your PR.